### DISCUSSION TITLE
Supporting Debian Raspberry Bookworm?

### DISCUSSION THREAD

**Discussion by reubenmiller** [silitics/rugix#2] [Ideas]
I've currently be trying to use use the latest Raspberry image which is based on Debian Bookworm and brings in support for Raspberry Pi 5 (rpi5), however the image does not boot when applying the image to a Raspberry Pi 4 (rpi4).

Looking at the related Raspberry Pi [release notes](https://downloads.raspberrypi.org/raspios_lite_armhf/release_notes.txt) for [2023-10-10-raspios-bookworm-arm64-lite.img.xz](https://downloads.raspberrypi.org/raspios_lite_arm64/images/raspios_lite_arm64-2023-10-10/2023-10-10-raspios-bookworm-arm64-lite.img.xz), it hints at potential root causes:

>* /boot mount point moved to /boot/firmware

In addition, the `rugpi-bakery bake` command includes the rpi4 eeproms, however I have been experimenting with excluding the eeproms from the image, however the image was still not bootable.

So here comes the main point questions/info (and hence why I'm posting in the discussion and not in the issues):

* Is the project open to PR's?
* I can offer testing support or information support about rp5 (if that would help)

**Comment by koehlma** [comment]
Regarding the EEPROM files: The reason for including those files is that we require the [A/B partition firmware features](https://github.com/raspberrypi/rpi-eeprom/blob/master/firmware-2711/release-notes.md#2022-10-18---tryboot-enhancements-for-ab-partition-booting---beta) which only have been introduced recently. On first boot, the files will lead the Pi 4 to update its firmware. Without those files, an off-the-shelf Pi 4 with an old firmware would not boot. I am not sure what the effects of these files on a Pi 5 are, though.

I just implemented an option for `rugpi-bakery.toml` to disable the placement of firmware files in the image (https://github.com/silitics/rugpi/commit/6a55cdbde32885e9e72e1873c3fd41544655adc5):

```toml
include_firmware = "none"
```

Regarding the switch to `/boot/firmware`: This does not make any difference. In fact, the `/boot` mount point [is not even included in `/etc/fstab`](https://github.com/silitics/rugpi/blob/66efba2fd762765eed8ad323734b6ee9eada132e/recipes/rugpi-ctrl/steps/01-run.sh#L6) right now. Also, Rugpi Bakery does correctly find the files as they are (still) on the first partition.

Presumably, the issues with Bookworm are because it comes with an `initramfs` by default, which interferes with Rugpi's initialization process. I just implemented a fix (https://github.com/silitics/rugpi/commit/0ce8cd06dbd744c0df9c0548e60ab3cfa5c732c3). Rugpi Bakery will now disable the respective option in `config.txt`.

I also pushed some other smaller fixes required for Bookworm to work. At least it works for me now. ;)

Can you confirm that these fixes solved your problems? (Please make sure to wait for the latest Docker images to become available and pull them in before trying it again.)

> Is the project open to PR's?

Yes, we very much welcome PRs.

> I can offer testing support or information support about rp5 (if that would help)

This would be great. We are also trying to get our hands on one.

**Reply by reubenmiller** [reply]
Sounds promising I'll test it out shortly.

Regarding the EEPROM for rp5, I'm assuming it would be destructive but that is only a guess, the files are slightly different, and the folder structure has changed, I was playing around with adding an option (like you did) [on my fork](https://github.com/silitics/rugpi/commit/0e588f80061a798135ec3c4f2d182967aa34dfad). Changes would be fairly trivial (and happy to create a PR as well).

> This would be great. We are also trying to get our hands on one.

Just a tip, since it looks like you're in Germany as well, the following site might have stock (one of the very few):
* https://shop.funk24.net/Raspberry-Pi/Raspberry-Pi-5/Pi-5-Bundles/

**Reply by reubenmiller** [reply]
To test it out I built the docker image locally, however building a new image with Debian bookworm worked perfectly on a Raspberry Pi 4 (5 still needs to be tested):

Below is the proof for Raspberry Pi 4:

<img width="1485" alt="image" src="https://github.com/silitics/rugpi/assets/3029781/4b9812da-f7f1-483d-9350-afec4a3eac72">

And the proof of which OS version it is running:

<img width="533" alt="image" src="https://github.com/silitics/rugpi/assets/3029781/94888808-6c11-4c15-8ce5-7fbab7580b37">

FYI: I'm testing rugpi integration with the project that I'm involved with [thin-edge.io](https://github.com/thin-edge/thin-edge.io)

**Reply by koehlma** [reply]
I am glad that it worked out.

Thanks for the tip with the Pi 5 – I just ordered one.

> FYI: I'm testing rugpi integration with the project that I'm involved with [thin-edge.io](https://github.com/thin-edge/thin-edge.io)

I know. Very exciting. Please let me know if you need anything to make this happen. In case you are interested, maybe we could also work out some sort of collaboration?

**Reply by reubenmiller** [reply]
And I can confirm the image works on Raspberry Pi 5.
